packages/client/shared/gqlIds/IntegrationRepoId.ts

```diff
@@ -21,38 +26,32 @@ type AzureDevOpsRepoIntegration = {
 
 type JiraServerRepoIntegration = {
   id: string
-  providerId: string
+  providerId: number
   key: string
   service: 'jiraServer'
 }
 
-export type RepoIntegration = {
-  id: string
-  providerId: string
-  service: Exclude<IntegrationProviderServiceEnum, 'jira' | 'jiraServer' | 'github' | 'azureDevOps'>
-}
+export type RepoIntegration =
+  | GitHubRepoIntegration
+  | JiraRepoIntegration
+  | JiraServerRepoIntegration
+  | AzureDevOpsRepoIntegration
+  | GitLabRepoIntegration
 
 const IntegrationRepoId = {
-  join: (
-    integration:
-      | GitHubRepoIntegration
-      | JiraRepoIntegration
-      | RepoIntegration
-      | JiraServerRepoIntegration
-      | AzureDevOpsRepoIntegration
-  ) => {
+  join: (integration: RepoIntegration) => {
     const {service} = integration
     switch (service) {
       case 'github':
         return integration.nameWithOwner
       case 'jira':
-        return JiraProjectId.join(integration.cloudId, integration.projectKey)
+        return JiraProjectId.join(integration.cloudId, integration.projectKey ?? integration.key!)
```

Comment:
`projectKey` doesn't exist in remote Jira projects, i.e. projects that we get directly from Jira's API. We use `IntegrationRepoId` for both remote Jira projects and our internal gql Jira projects, so I suggest checking for `projectKey` and `key`